daemons/attrd/attrd_utils.c

```diff
@@ -375,10 +381,43 @@ free_action(gpointer data)
 {
     struct confirmation_action *action = (struct confirmation_action *) data;
     g_list_free_full(action->respondants, free);
+    mainloop_timer_del(action->timer);
     free(action);
     action = NULL;
 }
 
+/* Remove the client from the expected_confirmations table if it times out before
+ * receiving all confirmations.  We set the timeout to 6000ms, which is 1000ms
+ * longer than the timeout used in send_attrd_request/connect_and_send_attrd_request
+ * (used by attrd_updater) just to give everything a moment to settle.
```

Comment:
Actually connect_and_send_attrd_request() can wait as long as 6 seconds before getting a successful connection, plus 5 seconds for pcmk__send_ipc_request(), and that doesn't include the time it takes the peers to respond, which is difficult to predict since their main loop could potentially have a backlog. I would go at least 15s.

Ideally we'd let the user override the timeout with a --timeout argument, but that might be a challenge to get to play nice with the IPC API model.

Reply:
This comment is probably just misleading.  The timeout of 6000ms here doesn't really have anything to do with the the client (attrd_updater or whatever).  It doesn't start until attrd_client_update is called, right before we broadcast the message to the other peer attrds in the cluster.  So it's really just a timer waiting for the peers to do everything and respond to the originating attrd.  I'll bump it to 15s anyway and change the comment.

It's not super critical the number is as low as it can be.  I just want a way of cleaning up dead things eventually.

Reply:
Fixed, but it doesn't look like the time out status code ever makes it back up to the command line tool.  It looks like as long as we were able to get some reply from the server, that's good enough and that's the error code that gets used.  Whatever actual status code is hiding in the XML reply doesn't matter.